nikola/post.py

```diff
@@ -92,176 +102,53 @@ def __init__(
-                            "TRANSLATIONS_PATTERN").format(self.source_path))
-        elif not self.translated_to:
-            raise Exception(("Cannot use {} (not a file, perhaps a broken "
-                            "symbolic link?)").format(self.source_path))
+        self._set_translated_to()
+        self._set_folders(destination, destination_base)
 
+        # Load default metadata
         default_metadata, default_used_extractor = get_meta(self, lang=None)
-
         self.meta = Functionary(lambda: None, self.default_lang)
         self.used_extractor = Functionary(lambda: None, self.default_lang)
         self.meta[self.default_lang] = default_metadata
         self.used_extractor[self.default_lang] = default_used_extractor
 
-        # Compose paths
-        if self.folder_base is not None:
-            # Use translatable destination folders
-            self.folders = {}
-            for lang in self.config['TRANSLATIONS'].keys():
-                if os.path.isabs(self.folder_base(lang)):  # Issue 2982
-                    self.folder_base[lang] = os.path.relpath(self.folder_base(lang), '/')
-                self.folders[lang] = os.path.normpath(os.path.join(self.folder_base(lang), self.folder_relative))
-        else:
-            # Old behavior (non-translatable destination path, normalized by scanner)
-            self.folders = {lang: self.folder_relative for lang in self.config['TRANSLATIONS'].keys()}
-        self.folder = self.folders[self.default_lang]
+        self._set_date(default_metadata)
 
-        if 'date' not in default_metadata and not use_in_feeds:
-            # For pages we don't *really* need a date
-            if self.config['__invariant__']:
-                default_metadata['date'] = datetime.datetime(2013, 12, 31, 23, 59, 59, tzinfo=tzinfo)
-            else:
-                default_metadata['date'] = datetime.datetime.utcfromtimestamp(
-                    os.stat(self.source_path).st_ctime).replace(tzinfo=dateutil.tz.tzutc()).astimezone(tzinfo)
-
-        # If time zone is set, build localized datetime.
-        try:
-            self.date = to_datetime(self.meta[self.default_lang]['date'], tzinfo)
-        except ValueError:
-            if not self.meta[self.default_lang]['date']:
-                msg = 'Missing date in file {}'.format(source_path)
-            else:
-                msg = "Invalid date '{0}' in file {1}".format(self.meta[self.default_lang]['date'], source_path)
-            LOGGER.error(msg)
-            raise ValueError(msg)
-
-        if 'updated' not in default_metadata:
-            default_metadata['updated'] = default_metadata.get('date', None)
-
-        self.updated = to_datetime(default_metadata['updated'], tzinfo)
-
-        if 'title' not in default_metadata or 'slug' not in default_metadata \
-                or 'date' not in default_metadata:
+        # These are the required metadata fields
+        if 'title' not in default_metadata or 'slug' not in default_metadata:
```

Comment:
if `date` was removed from the required metadata set, then maybe we could remove it from the message as well?

Reply:
Hmmm I am not sure I did the right thing here. Testing it.

Reply:
Confirmed, this is ok, it fails like this without ever reaching that message:

```
[2020-04-13T15:40:35Z] ERROR: Nikola: ValueError: Missing date in file posts/foo.rst
```